apps/go-mod-validator/src/deps.ts

```diff
@@ -0,0 +1,160 @@
+ *
+ *  type Module struct {
+ *        Path       string        // module path
+ *        Query      string        // version query corresponding to this version
+ *        Version    string        // module version
+ *        Versions   []string      // available module versions
+ *        Replace    *Module       // replaced by this module
+ *        Time       *time.Time    // time version was created
+ *        Update     *Module       // available update (with -u)
+ *        Main       bool          // is this the main module?
+ *        Indirect   bool          // module is only indirectly needed by main module
+ *        Dir        string        // directory holding local copy of files, if any
+ *        GoMod      string        // path to go.mod file describing module, if any
+ *        GoVersion  string        // go version used in module
+ *        Retracted  []string      // retraction information, if any (with -retracted or -u)
+ *        Deprecated string        // deprecation message, if any (with -u)
+ *        Error      *ModuleError  // error loading module
+ *        Origin     any           // provenance of module
+ *        Reuse      bool          // reuse of old module info is safe
+ *    }
+ *
+ *    type ModuleError struct {
+ *        Err string // the error itself
+ *    }
+ *
+ */
+interface GoMod {
+  Path: string;
+  Query: string;
+  Version: string;
+  Versions: string[];
+  Replace?: GoMod;
+  Time: Date | null;
+  Update?: GoMod;
+  Main: boolean;
+  Indirect: boolean;
+  Dir: string;
+  GoMod: string;
+  GoVersion: string;
+  Retracted: string[];
+  Deprecated: string;
+  Error: ModuleError | null;
+  Origin: any;
+  Reuse: boolean;
+}
+
+interface ModuleError {
+  Err: string;
+}
+export interface GoModule {
+  /**
+   * Module path
+   */
+  path: string;
+  /**
+   * Module version
+   */
+  version: string;
+}
+
+function parseGoModListOutput(jsonStr: string): GoMod[] {
+  /*
+   * {
+   *   "Path": "github.com/smartcontractkit/libocr",
+   *   "Version": "v0.0.0-20240419185742-fd3cab206b2c"
+   * }
+   * {
+   *   "Path": "github.com/smartcontractkit/go-proxy",
+   *   "Version": "v0.1.0"
+   * }
+   */
+
+  // store parsed objects
+  const goModules: GoMod[] = [];
+
+  // keep concatenating lines until the object is complete
+  let objStr = "";
+
+  // keep track of the depth of the object
+  let objDepth = 0;
+
+  // Process
+  for (const l of jsonStr.split("\n")) {
+    if (l.includes("{")) {
+      objDepth++;
+    }
+    if (l.includes("}")) {
+      objDepth--;
+    }
+
+    objStr += l;
+    if (objDepth === 0 && objStr) {
+      goModules.push(JSON.parse(objStr));
+      objStr = "";
+    }
+  }
+
+  return goModules;
+}
+
+/**
+ * Gets the paths of all go.mod files in the specified directory.
+ *
+ * @param goModDir The directory to search for go.mod files.
+ *
+ * @throws If no go.mod files are found or an error occurs.
+ */
+async function getGoModFiles(goModDir: string): Promise<string[]> {
```

Comment:
IMO it is not clear that this method is globbing recursively for go.mods. Rename some stuff?